### PR TITLE
Alp - 9 - Tab Component

### PR DESCRIPTION
**Pull request by Leonlui**

### CODE REVIEW

File: aunt-leahs-app/src/components/tab/tab.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AppBar from '@material-ui/core/AppBar';
+import Tabs from '@material-ui/core/Tabs';
+import Tab from '@material-ui/core/Tab';
+import Typography from '@material-ui/core/Typography';
+import Box from '@material-ui/core/Box';
+
+//How to Call this class from another class:
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/tab/tab.js:9]: Just write "usage"

File: aunt-leahs-app/src/components/tab/tab.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        }
+    }
+    handleChange = (event, newValue) => {
+        this.setState({ value: newValue })
+    };
+    
+    render () {
+        return  <div className={this.props.name || "TabComponent"}>
+        <AppBar position="static">
+            <Tabs value={this.state.value} onChange={this.handleChange} aria-label= "simple tabs example">
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/tab/tab.js:34]: Put attributes on separate lines for readability

File: aunt-leahs-app/src/components/tab/tab.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                {this.state.items.map((item, index) =>
+                    <Tab label={item.title} {...a11yProps(index)} />
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/tab/tab.js:36]: What are A one one Y props? :thinking: :thinking:

File: aunt-leahs-app/src/components/tab/tab.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+TabPanel.propTypes = {
+    children: PropTypes.node,
+    index: PropTypes.any.isRequired,
+    value: PropTypes.any.isRequired,
+};
+
+function a11yProps(index) {
+    return {
+        id: `simple-tab-${index}`,
+        'aria-controls': `simple-tabpanel-${index}`,
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/tab/tab.js:75]: Don't need quotes around "aria-controls". But if you want them, put them around all of the keys, and use double quotes, not single.

File: aunt-leahs-app/src/components/tab/tab.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            value: 0,
+            items: this.props.items || [],
+        }
+    }
+    handleChange = (event, newValue) => {
+        this.setState({ value: newValue })
+    };
+    
+    render () {
+        return  <div className={this.props.name || "TabComponent"}>
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/tab/tab.js:32]: Why are there so many spaces?

File: aunt-leahs-app/src/components/tab/tab.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+function TabPanel(props) {
+    const { children, value, index, ...other } = props;
+
+    return (
+        <Typography
+            component="div"
+            role="tabpanel"
+            hidden={value !== index}
+            id={`simple-tabpanel-${index}`}
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/tab/tab.js:57]: I feel like `"simple-tabpanel-" + index` is more readable but maybe that's just me

File: aunt-leahs-app/src/components/tab/tab.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    );
+}
+
+TabPanel.propTypes = {
+    children: PropTypes.node,
+    index: PropTypes.any.isRequired,
+    value: PropTypes.any.isRequired,
+};
+
+function a11yProps(index) {
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/tab/tab.js:72]: What's a11y? Why is this named this way?

File: aunt-leahs-app/src/components/tab/tab.js (hunk 2 above)
Comment (justinkkwan) [inline aunt-leahs-app/src/components/tab/tab.js:34]: Why is one of your labels called `example`?

File: aunt-leahs-app/src/components/tab/tab.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    return {
+        id: `simple-tab-${index}`,
+        'aria-controls': `simple-tabpanel-${index}`,
+    };
+}
+
+export default TabComponent
```
Comment (justinkkwan) [inline aunt-leahs-app/src/components/tab/tab.js:79]: \n

**Review by justinkkwan — CHANGES_REQUESTED** [review]
:-1:

File: aunt-leahs-app/src/components/tab/tab.js (hunk 5 above)
Comment (Leonlui) [reply aunt-leahs-app/src/components/tab/tab.js:32]: theres literally only one space...

File: aunt-leahs-app/src/components/tab/tab.js (hunk 6 above)
Comment (Leonlui) [reply aunt-leahs-app/src/components/tab/tab.js:57]: its just you

File: aunt-leahs-app/src/components/tab/tab.js (hunk 5 above)
Comment (justinkkwan) [reply aunt-leahs-app/src/components/tab/tab.js:32]: `return <SPACE> <SPACE> <div className=...`

File: aunt-leahs-app/src/components/tab/tab.js (hunk 8 above)
Comment (Leonlui) [reply aunt-leahs-app/src/components/tab/tab.js:79]: There is a new line, it there isnt then github will tell you

File: aunt-leahs-app/src/components/tab/tab.js (hunk 4 above)
Comment (Leonlui) [reply aunt-leahs-app/src/components/tab/tab.js:75]: i do need quotes around them

**Comment by Leonlui** [comment]
Made significant changes to the styling

**Pull request closed without merging**